### PR TITLE
adding form JSON generator

### PR DESCRIPTION
**Pull request by alagroy-42**
This is the form JSON Generator in `Lizmap\Project\QgisProject`. If you build the app, you will be able to create the JSON and check it. It will be located in `lizmap/modules/lizmap/forms/project.layer.form.json`. However, the edition part will not work because of the namespace changing and I didn't modify qgisForm* classes to work with the JSON, it will be for another PR.

### CODE REVIEW

**Review by rldhont — APPROVED** [review]

**Review by laurentj — CHANGES_REQUESTED** [review]
you must not store generated files into lizmap directories. Store them into the temp directory.

You should also explain what you're changing into commit messages. I'm not sure what you are doing, since there are too much changes.

**Comment by alagroy-42** [comment]
About the json files storing location, I did a method to get it in AppContext in next PR, so I'll fix it on #1844 to avoid conflicts. Sorry for that.

File: lizmap/modules/lizmap/lib/Project/QgisProject.php [hunk 1]
```diff
@@ -1362,4 +1384,234 @@ protected function readLayers($xml)
...
+    protected function getEditType($layerXml)
+    {
+        $edittypes = $layerXml->edittypes;
+        $editTab = array();
+
+        foreach ($edittypes as $edittype) {
+            $attributes = $edittype->attributes();
+            $editTab[$attributes->name] = array();
+            $editTab[$attributes->name]['eddittype'] = $edittype;
+            if (property_exists($edittype->attributes(), 'widgetv2type')) {
```
Comment (laurentj) [inline lizmap/modules/lizmap/lib/Project/QgisProject.php:1476]: an original comment is missing :  `// New QGIS 2.4 edittypes : use widgetv2type property`

**Review by laurentj — APPROVED** [review]
Looks good to me. I will apply request changes

**Comment by laurentj** [comment]
Merged by hand.

**Pull request closed without merging**